cms/djangoapps/contentstore/views/import_export.py

```diff
@@ -2,485 +2,166 @@
-
-    If the tar.gz file has been requested but the export operation fails, an HTML page will be returned
-    which describes the error.
     """
-    course_key = CourseKey.from_string(course_key_string)
-    export_url = reverse_course_url('export_handler', course_key)
-    if not has_course_author_access(request.user, course_key):
-        raise PermissionDenied()
+    error = request.GET.get("error", None)
+    error_message = request.GET.get("error_message", None)
+    failed_module = request.GET.get("failed_module", None)
+    unit = request.GET.get("unit", None)
+    parent = request.GET.get("parent", None)
 
-    if isinstance(course_key, LibraryLocator):
-        courselike_module = modulestore().get_library(course_key)
-        context = {
-            'context_library': courselike_module,
-            'courselike_home_url': reverse_library_url("library_handler", course_key),
-            'library': True
-        }
+    courselike_key = CourseKey.from_string(course_key_string)
+    library = isinstance(courselike_key, LibraryLocator)
+    if library:
+        successful_url = reverse_library_url("library_handler", courselike_key)
+        courselike_module = modulestore().get_library(courselike_key)
+        context_name = "context_library"
     else:
-        courselike_module = modulestore().get_course(course_key)
-        context = {
-            'context_course': courselike_module,
-            'courselike_home_url': reverse_course_url("course_handler", course_key),
-            'library': False
-        }
+        successful_url = reverse_course_url("course_handler", courselike_key)
+        courselike_module = modulestore().get_course(courselike_key)
+        context_name = "context_course"
 
-    context['export_url'] = export_url + '?_accept=application/x-tgz'
+    if not has_course_author_access(request.user, courselike_key):
+        raise PermissionDenied()
 
-    # an _accept URL parameter will be preferred over HTTP_ACCEPT in the header.
-    requested_format = request.REQUEST.get('_accept', request.META.get('HTTP_ACCEPT', 'text/html'))
+    export_url = reverse(
+        "course_import_export_handler",
+        kwargs={
+            "course_key_string": unicode(courselike_key),
+        }
+    ) + "?accept=application/x-tgz"
+
+    export_url += "&{0}".format(
+        urlencode({
+            "redirect": reverse_course_url(
+                "export_handler",
+                unicode(courselike_key)
+            )
+        })
+    )
 
-    if 'application/x-tgz' in requested_format:
+    if parent:
         try:
-            tarball = create_export_tarball(courselike_module, course_key, context)
-        except SerializationError:
-            return render_to_response('export.html', context)
-        return send_tarball(tarball)
-
-    elif 'text/html' in requested_format:
-        return render_to_response('export.html', context)
+            parent = CourseKey.from_string(parent)
```

Comment:
In general, please avoid using the same variable to hold different types of objects. It can make it confusing to debug later on, particularly if a variable changes its type in a conditional and may be different things further on in the code. Please just use a new var name, like `parent_key`.